### DISCUSSION TITLE
Find an equal number of clusters between scRNA-seq and scATAC-seq experiments

### DISCUSSION THREAD

**Discussion by RegnerM2015** [satijalab/seurat#5030] [Q&A] [answered]
Hi @timoast,

## Background
I have a complex study design that involves cluster label transferring between scRNA-seq and scATAC-seq. Considering the following hypothetical scenario:

The goal is to transfer labels between matched scRNA-seq and scATAC-seq datasets for a total of 10 tissue samples.

- We combine the 10 scRNA-seq datasets using Harmony integration to create one multi-sample scRNA-seq dataset and recluster the dataset with Louvain clustering (note that the underlying data is untouched b/c Harmony operates on low dim. embeddings).

- We combine the 10 scATAC-seq datasets using Harmony integration to create one multi-sample scATAC-seq dataset.

- Finally, we use CCA to transfer labels from the multi-sample scRNA-seq dataset to the multi-sample scATAC-seq dataset.

## Problem
My issue is that not all of the cluster labels from scRNA-seq are transferred over to the corresponding scATAC-seq dataset for a given resolution. In other words, the scATAC-seq cells are not mapping to all of the scRNA-seq clusters. To help illustrate this concept, here is a visual:
<img width="1114" alt="Screen Shot 2021-09-01 at 5 24 31 PM" src="https://user-images.githubusercontent.com/54336249/131747497-ff8ef53d-4c7c-46f7-a46e-d207797f7211.png">

## Potential solution
As the resolution in scRNA-seq clustering increases, the 'recovery rate' of scRNA-seq clusters in scATAC-seq decreases. I have decided to iterate through several resolutions of clustering in scRNA-seq, and for each iteration, I transfer labels to the corresponding scATAC-seq dataset and record a few statistics. Below is a line chart that shows how this 'recovery rate' decays with increasing resolution: 

The y axis on the left represents the number of inferred clusters in scATAC-seq (after filtering cells for prediction score >=0.5) divided by the number of clusters in scRNA-seq for that given resolution. The y axis on the right represents the same proportion, but restricted to only epithelial clusters. 

<img width="685" alt="Screen Shot 2021-09-01 at 5 31 47 PM" src="https://user-images.githubusercontent.com/54336249/131748066-28854169-bc63-40d3-a99d-c1a353425a74.png">

## Thoughts? Suggestions?

Which resolution is most preferable here for scRNA-seq? One may expect an equal number of cell populations between two data modalities. In other words, I would expect that a robust clustering scheme identified in scRNA-seq should also be conserved in the corresponding scATAC-seq data. 

I think that the multi-sample scale and Harmony correction is complicating this task. Ideally, we want the same number of clusters conserved across data modalities, but we also want to acknowledge rare cell type populations at higher resolutions of clustering in scRNA-seq. 

How would you proceed? Any feedback or comments would be greatly appreciated!

**Comment by timoast** [comment, marked as answer]
See https://github.com/timoast/signac/discussions/770
